lib/Target/Lattigo/LattigoEmitter.cpp

```diff
@@ -228,6 +232,66 @@ LogicalResult LattigoEmitter::printOperation(func::CallOp op) {
   return success();
 }
 
+LogicalResult LattigoEmitter::printOperation(affine::AffineForOp op) {
+  if (!op.hasConstantBounds()) {
+    return emitError(op.getLoc(), "Only constant bounds are supported");
+  }
+
+  for (auto i = 0; i < op.getNumRegionIterArgs(); ++i) {
+    // Assign the loop's results to use as initial iter args.
+    Value result = op.getResults()[i];
+    Value operand = op.getOperands()[i];
+    Value iterArg = op.getRegionIterArgs()[i];
+
+    if (variableNames->contains(result) && variableNames->contains(operand) &&
+        variableNames->getNameForValue(result) ==
+            variableNames->getNameForValue(operand)) {
+      // This occurs in cases where the loop is inserting into a tensor and
+      // passing it along as an inter arg.
+      continue;
+    }
+
+    os << getName(iterArg) << " := " << getName(operand) << "\n";
+  }
+
+  os << llvm::formatv("for {0} := {1}; {0} < {2}; {0} += 1 {{\n",
```

Comment:
`+= 1` seems hard encoding but it is fine for now.

Reply:
This is a good point, let me fix it. Also I think I could ensure all the iteration indices are `int64` because elsewhere `index` type is cast to `int64`.

Reply:
Fixed and force pushed